app/graphql/presenters/overriden_ansible_variable_presenter.rb

```diff
@@ -2,18 +2,32 @@ module Presenters
   class OverridenAnsibleVariablePresenter
     attr_reader :ansible_variable
 
-    delegate :id, :key, :description, :override?,
-             :parameter_type, :hidden_value?, :omit, :required,
-             :validator_type, :validator_rule, :default_value,
-             :ansible_role, :current_value, :to => :ansible_variable
+    delegate :id, :key, :description,
+             :parameter_type, :omit, :required,
+             :validator_type, :validator_rule,
+             :ansible_role, :to => :ansible_variable
+
+    def default_value
+      ansible_variable.editable_by_user? ? ansible_variable.default_value : '*****'
+    end
+
+    def hidden_value
+      ansible_variable.hidden_value?
+    end
+
+    def override
+      ansible_variable.override?
+    end
 
     def initialize(ansible_variable, override_resolver)
       @ansible_variable = ansible_variable
       @override_resolver = override_resolver
     end
 
     def current_value
-      @override_resolver.resolve @ansible_variable
+      resolved = @override_resolver.resolve @ansible_variable
+      resolved[:value] = '*****' unless ansible_variable.editable_by_user?
+      resolved
```

Comment:
We need to check if the resolved value is not nil, otherwise we will get an error when trying to set the value.